### PR TITLE
feat(plasma-tokens,plama-tokens-utils): Update theme name convention

### PR DESCRIPTION
**Pull request by neretin-trike**
Согласно новой конвенции именования тем, принимается следующий формат:
`<название_темы>_<вид_темы>__<модификаторы_темы>` или `<название_темы>__<модификаторы_темы>`.

Так сейчас выглядит список тем с поддержкой обратной совместимости:

Screenshot 2023-04-11 at 23 55 34

В случае использовании старых названий тем они будут помечены как `deprecated`, с комментарием, какое название вместо этого лучше использовать:

Screenshot 2023-04-11 at 23 55 21

Так же для улучшения консистентности, для тем, которые используются в canvas приложениях, был добавлен префикс `salutejs`.

:baby_chick: Download canary assets:

[colors--canary.475.4680213255.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.475.4680213255.xml)  
[color_sberHealth_ios-swift--canary.475.4680213255.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.475.4680213255.swift)  
[color_sberHealth_kotlin--canary.475.4680213255.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.475.4680213255.kt)  
[color_sberHealth_react-native--canary.475.4680213255.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.475.4680213255.ts)  
[color_sbermarket_business_ios-swift--canary.475.4680213255.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_ios-swift--canary.475.4680213255.swift)  
[color_sbermarket_business_kotlin--canary.475.4680213255.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_kotlin--canary.475.4680213255.kt)  
[color_sbermarket_business_react-native--canary.475.4680213255.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_react-native--canary.475.4680213255.ts)  
[color_sbermarket_ios-swift--canary.475.4680213255.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.475.4680213255.swift)  
[color_sbermarket_kotlin--canary.475.4680213255.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.475.4680213255.kt)  
[color_sbermarket_metro_ios-swift--canary.475.4680213255.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_ios-swift--canary.475.4680213255.swift)  
[color_sbermarket_metro_kotlin--canary.475.4680213255.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_kotlin--canary.475.4680213255.kt)  
[color_sbermarket_metro_react-native--canary.475.4680213255.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_react-native--canary.475.4680213255.ts)  
[color_sbermarket_react-native--canary.475.4680213255.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.475.4680213255.ts)  
[color_sbermarket_selgros_ios-swift--canary.475.4680213255.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_ios-swift--canary.475.4680213255.swift)  
[color_sbermarket_selgros_kotlin--canary.475.4680213255.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_kotlin--canary.475.4680213255.kt)  
[color_sbermarket_selgros_react-native--canary.475.4680213255.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_react-native--canary.475.4680213255.ts)  
[color_sbermarket_winelab_ios-swift--canary.475.4680213255.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_winelab_ios-swift--canary.475.4680213255.swift)  
[color_sbermarket_winelab_kotlin--canary.475.4680213255.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_winelab_kotlin--canary.475.4680213255.kt)  
[color_sbermarket_winelab_react-native--canary.475.4680213255.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_winelab_react-native--canary.475.4680213255.ts)  
[color_sberprime_ios-swift--canary.475.4680213255.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.475.4680213255.swift)  
[color_sberprime_kotlin--canary.475.4680213255.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.475.4680213255.kt)  
[color_sberprime_react-native--canary.475.4680213255.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.475.4680213255.ts)  
[PlasmaTokensColor--canary.475.4680213255.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.475.4680213255.swift)  
[shadow_sbermarket_react-native--canary.475.4680213255.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.475.4680213255.ts)  
[typo_mage_ios-swift--canary.475.4680213255.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.475.4680213255.swift)  
[typo_mage_kotlin--canary.475.4680213255.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.475.4680213255.kt)  
[typo_mage_react-native--canary.475.4680213255.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.475.4680213255.ts)  
[typo_plasma_ios-swift--canary.475.4680213255.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.475.4680213255.swift)  
[typo_plasma_kotlin--canary.475.4680213255.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.475.4680213255.kt)  
[typo_plasma_react-native--canary.475.4680213255.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.475.4680213255.ts)  
[typo_ruler_ios-swift--canary.475.4680213255.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.475.4680213255.swift)  
[typo_ruler_kotlin--canary.475.4680213255.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.475.4680213255.kt)  
[typo_ruler_react-native--canary.475.4680213255.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.475.4680213255.ts)  
[typo_sage_ios-swift--canary.475.4680213255.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.475.4680213255.swift)  
[typo_sage_kotlin--canary.475.4680213255.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.475.4680213255.kt)  
[typo_sage_react-native--canary.475.4680213255.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.475.4680213255.ts)  
[typo_sbermarket_ios-swift--canary.475.4680213255.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.475.4680213255.swift)  
[typo_sbermarket_kotlin--canary.475.4680213255.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.475.4680213255.kt)  
[typo_sbermarket_react-native--canary.475.4680213255.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.475.4680213255.ts)  
[typo_soulmate_ios-swift--canary.475.4680213255.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.475.4680213255.swift)  
[typo_soulmate_kotlin--canary.475.4680213255.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.475.4680213255.kt)  
[typo_soulmate_react-native--canary.475.4680213255.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.475.4680213255.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.175.0-canary.475.4680213255.0
  npm install @salutejs/plasma-core@1.109.0-canary.475.4680213255.0
  npm install @salutejs/plasma-hope@0.33.0-canary.475.4680213255.0
  npm install @salutejs/plasma-icons@1.138.0-canary.475.4680213255.0
  npm install @salutejs/plasma-temple@1.151.0-canary.475.4680213255.0
  npm install @salutejs/plasma-tokens-b2b@1.15.0-canary.475.4680213255.0
  npm install @salutejs/plasma-tokens-b2c@0.24.0-canary.475.4680213255.0
  npm install @salutejs/plasma-tokens-web@1.30.0-canary.475.4680213255.0
  npm install @salutejs/plasma-tokens@1.45.0-canary.475.4680213255.0
  npm install @salutejs/plasma-ui@1.183.0-canary.475.4680213255.0
  npm install @salutejs/plasma-web@1.200.0-canary.475.4680213255.0
  npm install @salutejs/plasma-cy-utils@0.52.0-canary.475.4680213255.0
  npm install @salutejs/plasma-sb-utils@0.107.0-canary.475.4680213255.0
  npm install @salutejs/plasma-tokens-android@2.52.0-canary.475.4680213255.0
  npm install @salutejs/plasma-tokens-ios-swift@2.52.0-canary.475.4680213255.0
  npm install @salutejs/plasma-tokens-utils@0.21.0-canary.475.4680213255.0
  # or 
  yarn add @salutejs/plasma-b2c@1.175.0-canary.475.4680213255.0
  yarn add @salutejs/plasma-core@1.109.0-canary.475.4680213255.0
  yarn add @salutejs/plasma-hope@0.33.0-canary.475.4680213255.0
  yarn add @salutejs/plasma-icons@1.138.0-canary.475.4680213255.0
  yarn add @salutejs/plasma-temple@1.151.0-canary.475.4680213255.0
  yarn add @salutejs/plasma-tokens-b2b@1.15.0-canary.475.4680213255.0
  yarn add @salutejs/plasma-tokens-b2c@0.24.0-canary.475.4680213255.0
  yarn add @salutejs/plasma-tokens-web@1.30.0-canary.475.4680213255.0
  yarn add @salutejs/plasma-tokens@1.45.0-canary.475.4680213255.0
  yarn add @salutejs/plasma-ui@1.183.0-canary.475.4680213255.0
  yarn add @salutejs/plasma-web@1.200.0-canary.475.4680213255.0
  yarn add @salutejs/plasma-cy-utils@0.52.0-canary.475.4680213255.0
  yarn add @salutejs/plasma-sb-utils@0.107.0-canary.475.4680213255.0
  yarn add @salutejs/plasma-tokens-android@2.52.0-canary.475.4680213255.0
  yarn add @salutejs/plasma-tokens-ios-swift@2.52.0-canary.475.4680213255.0
  yarn add @salutejs/plasma-tokens-utils@0.21.0-canary.475.4680213255.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-475/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-475/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-475/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-475/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-475/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-475/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Review by Yakutoc — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-475/

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-475/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-475/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-475/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-475/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-475/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-475/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-475/temple-storybook/

**Review by kayman233 — APPROVED** [review]

File: utils/plasma-tokens-utils/src/generators/generateColorThemes.ts [hunk 1]
```diff
@@ -9,15 +9,25 @@ import { generateFile } from './generateFile';
...
 
     for (const [fileName, themeItem] of Object.entries(colorThemes)) {
         const { fromData, ...theme } = themeItem;
-
         const themeData = extractTokenData(theme);
-        indexContent += `export { ${fileName} } from './${fileName}';\n`;
+        const [deprecated, deprecatedThemeName] = withDeprecated ? getDeprecatedVars(fileName) : ['', ''];
+
+        const fileNames = [fileName];
+        if (withDeprecated && deprecated) {
```
Comment (Yeti-or) [inline utils/plasma-tokens-utils/src/generators/generateColorThemes.ts:26]: 👍

File: utils/plasma-tokens-utils/src/utils/upperFirstLetter.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export const upperFirstLetter = (value: string) => `${value[0].toLocaleUpperCase()}${value.slice(1)}`;
```
Comment (Yeti-or) [inline utils/plasma-tokens-utils/src/utils/upperFirstLetter.ts:1]: надо будет как-нибудь собрать микробенч на самый быстрый способ собрать строку с первой заглавной
у меня вот такой есть вариант: `str.replace(/^./, (x) => x.toUpperCase())`

**Review by Yeti-or — APPROVED** [review]

File: utils/plasma-tokens-utils/src/utils/upperFirstLetter.ts (hunk 2 above)
Comment (neretin-trike) [reply utils/plasma-tokens-utils/src/utils/upperFirstLetter.ts:1]: хм, кажется, что с регексом будет быстрее работать. Попробую этот вариант юзать

**Pull request merged**